### PR TITLE
Feature/auth update

### PR DESCRIPTION
**Pull request by Michiocre**
Updated to jira oauth 2.0
Added Bugs page in the presentation
Issues are now sorted by Assignees

### CODE REVIEW

File: pages/api/oauth/callback.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  if (response.status !== 200) {
+    const error = await response.json();
+    console.log(error);
+    return res.status(response.status).send(error);
+  }
+
+  const { access_token, expires_in } = await response.json();
+
+  // const cloudResponse = await fetch(
```
Comment (mattcrn) [inline pages/api/oauth/callback.js:54]: I guess we can remove that

File: pages/api/sprints/active.js [hunk 2]
```diff
@@ -49,24 +40,27 @@ const handler = async (req, res) => {
...
       );
 
-    const data = [...apiData, ...dbData].filter((sprint) =>
-      favourites.some(
-        (fav) =>
-          fav.id === sprint.originBoardId ||
-          fav.id === parseInt(sprint.boardId),
-      ),
-    );
+    // const data = [...apiData, ...dbData].filter((sprint) =>
```
Comment (mattcrn) [inline pages/api/sprints/active.js:51]: Is this not needed anymore? then lets remove it

File: .env.example [hunk 3]
```diff
@@ -5,4 +5,9 @@ DATABASE_NAME=sprint-slides
 DATABASE_USER=root
 DATABASE_PASSWORD=root
 
-SECRET_COOKIE_PASSWORD=abcdefghijklmnopqrstuvw
\ No newline at end of file
+SECRET_COOKIE_PASSWORD=abcdefghijklmnopqrstuvwxyz0123456789
+
+API_URL='https://api.atlassian.com/ex/jira/'
```
[the remark this answers is not in the document]
Comment (Michiocre) [reply .env.example:10]: See https://github.com/towa-digital/sprint-slides-generator/pull/15/commits/7e291508d3f67e238e0c4478688b939330a01435

File: pages/api/sprints/active.js (hunk 2 above)
Comment (Michiocre) [reply pages/api/sprints/active.js]: See https://github.com/towa-digital/sprint-slides-generator/pull/15/commits/14875f7b2f7cb87c0a537c182b4e1386ebd0d12f

File: pages/api/oauth/callback.js (hunk 1 above)
Comment (Michiocre) [reply pages/api/oauth/callback.js]: See https://github.com/towa-digital/sprint-slides-generator/pull/15/commits/14875f7b2f7cb87c0a537c182b4e1386ebd0d12f

File: utils/jiraApi.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Version2Client, Agile } from 'jira.js';
+
+const cloudId = '918d98df-4747-4c9d-b197-d7d658581d27';
```
Comment (Gabsii) [inline utils/jiraApi.js:3]: move to .env if this is sensitive data

File: utils/sprintMetrics.js [hunk 5]
```diff
@@ -1,24 +1,47 @@
...
+  if (bugs) {
+    return bugs.reduce((acc, bug) => (acc += bug.fields.customfield_10028), 0);
+  }
+
+  if (others) {
+    return others.reduce(
+      (acc, other) => (acc += other.fields.customfield_10028),
+      0,
+    );
+  }
```
Comment (Gabsii) [inline utils/sprintMetrics.js:33]: if we have bugs and other types of tickets we won't return the others correctly this way

File: pages/api/oauth/redirect.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+export default async (req, res) => {
+  const YOUR_USER_BOUND_VALUE = crypto
+    .createHash('sha256', process.env.SECRET_COOKIE_PASSWORD)
+    .digest('hex');
+
+  //%20offline_access add this scope to enable refresh token
+  res.setHeader(
+    'Location',
+    `https://auth.atlassian.com/authorize?audience=api.atlassian.com&client_id=V27DQrIWINbWI9VuSOcxI6QVPptPQsks&scope=read%3Asprint%3Ajira-software%20read%3Agroup%3Ajira%20read%3Aapplication-role%3Ajira%20read%3Auser%3Ajira%20read%3Aavatar%3Ajira%20read%3Aboard-scope.admin%3Ajira-software%20read%3Aboard-scope%3Ajira-software%20read%3Aissue-details%3Ajira%20read%3Ajql%3Ajira%20offline_access&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Foauth%2Fcallback&state=${YOUR_USER_BOUND_VALUE}&response_type=code&prompt=consent`,
```
Comment (Gabsii) [inline pages/api/oauth/redirect.js:11]: move client_id redirect_id to .env

File: pages/api/oauth/redirect.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const crypto = require('crypto');
+
+export default async (req, res) => {
+  const YOUR_USER_BOUND_VALUE = crypto
+    .createHash('sha256', process.env.SECRET_COOKIE_PASSWORD)
+    .digest('hex');
+
+  //%20offline_access add this scope to enable refresh token
```
Comment (Gabsii) [inline pages/api/oauth/redirect.js:8]: remove comment

File: pages/boards.jsx [hunk 8]
```diff
@@ -167,13 +167,23 @@ const Boards = ({ boards, user, authToken, favourites, errors }) => {
 export const getServerSideProps = withSession(async function ({ req, res }) {
   let errors = [];
   const user = sessionData(req, res, 'user');
-  const authToken = sessionData(req, res, 'authToken');
+  let authToken = sessionData(req, res, 'authToken');
+  let expiresAt = sessionData(req, res, 'expiresAt');
```
Comment (Gabsii) [inline pages/boards.jsx:171]: we don't refresh so we don't need this in the session

File: pages/dashboard.jsx [hunk 9]
```diff
@@ -102,12 +102,23 @@ const Dashboard = ({ user, favourites, activeSprints, authToken, errors }) => {
 export const getServerSideProps = withSession(async function ({ req, res }) {
   let errors = [];
   const user = sessionData(req, res, 'user');
-  const authToken = sessionData(req, res, 'authToken');
+  let authToken = sessionData(req, res, 'authToken');
+  let expiresAt = new Date(sessionData(req, res, 'expiresAt'));
```
Comment (Gabsii) [inline pages/dashboard.jsx:106]: same here. 
IMO the user should just be redirected if something fails in regards to the authtoken

File: pages/index.jsx [hunk 10]
```diff
@@ -3,6 +3,7 @@ import { Page } from '@zeit-ui/react';
 
 import withSession from '@utils/session';
 import LoginForm from '@components/LoginForm';
+import { Button } from '@zeit-ui/react';
```
Comment (Gabsii) [inline pages/index.jsx:6]: we're not using a button here

File: pages/sprint/[slug].jsx [hunk 11]
```diff
@@ -111,6 +121,8 @@ const Sprint = ({ user, currentSprint, data }) => {
   const completedPoints = completedStoryPoints(stories, bugs, others);
   const pointsInReview = pointsNeedReview(data);
 
+  memoCurrentSprint.title = memoCurrentSprint.sprintName + ' | SprintGenerator'
```
Comment (Gabsii) [inline pages/sprint/[slug].jsx:124]: this should happen in the use memo itself

**Review by Gabsii — CHANGES_REQUESTED** [review]

File: pages/boards.jsx (hunk 8 above)
Comment (Michiocre) [reply pages/boards.jsx:171]: I still use expiresAt to redirect the user before a invalid request is made

File: pages/boards.jsx [hunk 12]
```diff
@@ -196,6 +186,15 @@ export const getServerSideProps = withSession(async function ({ req, res }) {
   });
   favouritesError && errors.push(favouritesError);
 
+  if (boards.code === 401) {
+    req.session.destroy();
+    res.writeHead(302, { Location: '/' });
+    res.end();
+    return {
+      props: {},
+    };
```
Comment (Gabsii) [inline pages/boards.jsx:195]: since this is used in multiple places - can we maybe move it to a function that takes req and res and returns the props?

**Review by Gabsii — APPROVED** [review]

**Pull request merged**